File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int len = 0;
        ListNode temp = head;
        while (temp != null) {
            temp = temp.next;
            len++;
        }
        
        temp = head;
        if (len - n == 0) {
            head = head.next;
            return head;
        }
        for (int i = 0; i < len - n - 1; i++) {
            temp = temp.next;
        }
        if (temp.next != null)
            temp.next = temp.next.next;
        else {
            temp.next = null;
        }
        return head;
    }
}

File: Data Structures & Algorithms/linked-list-cycle-detection/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        if not head:
            return False
        slow = head
        fast = head.next
        while fast:
            if slow == fast:
                return True
            fast = fast.next
            if slow == fast:
                return True
            if fast:
                fast = fast.next
            if slow == fast:
                return True

            slow = slow.next
        
        return False